lang/cs/Org.Apache.REEF.IMRU.Tests/JobLifecycleManagerTest.cs

```diff
@@ -0,0 +1,210 @@
+                .FirstOrDefault();
+
+            AssertCancelEvent(observer, true, expectedMessage);
+        }
+
+        [Fact]
+        public void JobLifeCyclemanger_SendsJobCancelledEventToMultiplyObservers()
+        {
+            string expectedMessage = "cancelled";
+            var observers = JobLifeCycleMangerEventTest(
+                detector: new DetectorStubStatic(true, expectedMessage));
+
+            foreach (var observer in observers)
+            {
+                AssertCancelEvent(observer, true, expectedMessage);
+            }
+        }
+
+        [Fact]
+        public void JobLifeCyclemanger_ChecksDetectorPeriodically()
+        {
+            string expectedMessage = "cancelled";
+            int isCancelledCheckCounter = 0;
+
+            var observer = JobLifeCycleMangerEventTest(
+                detector: new DetectorStubStatic(true, expectedMessage, testAction: () => { isCancelledCheckCounter++; }),
+                signalCheckPeriodSec: 1,
+                waitForEventPeriodSec: 6)
+                .FirstOrDefault();
+
+            Assert.True(isCancelledCheckCounter >= 5, "Expected 5+ IsCancelled checks in 6 sec (check interval = 1 sec). Actual check counter: " + isCancelledCheckCounter);
+            AssertCancelEvent(observer, true, expectedMessage);
+        }
+
+        [Fact]
+        public void JobLifeCyclemanger_NoSignal_DoesNotSendEvent()
+        {
+            var observer = JobLifeCycleMangerEventTest(
+                detector: new DetectorStubStatic(false))
+                .FirstOrDefault();
+
+            AssertCancelEvent(observer, false);
+        }
+
+        [Fact]
+        public void JobLifeCyclemanger_DetectorNull_DoesNotSendEvent()
+        {
+            var observer = JobLifeCycleMangerEventTest(
+                detector: null)
+                .FirstOrDefault();
+
+            AssertCancelEvent(observer, false);
+        }
+
+        [Fact]
+        public void JobLifeCyclemanger_NoObservers_DoesNotCheckForSignal()
+        {
+            int isCancelledCheckCounter = 0;
+
+            var observer = JobLifeCycleMangerEventTest(
+                detector: new DetectorStubStatic(true, "cancelled", testAction: () => { isCancelledCheckCounter++; }),
+                subscribeObserver: false,
+                signalCheckPeriodSec: 1,
+                waitForEventPeriodSec: 6)
+                .FirstOrDefault();
+
+            Assert.True(isCancelledCheckCounter == 0, "Expected no checks for cancellation if there are no subscribers. Actual check counter: " + isCancelledCheckCounter);
+            AssertCancelEvent(observer, false);
+        }
+
+        private IEnumerable<TestObserver> JobLifeCycleMangerEventTest(
+            IJobCancelledDetector detector,
+            bool subscribeObserver = true,
+            int observerCount = 1,
+            int signalCheckPeriodSec = 1,
+            int waitForEventPeriodSec = 2)
+        {
+            var manager = Activator.CreateInstance(
```

Comment:
the constructor is private by prev comments

---
In reply to: [85997021](https://github.com/apache/reef/pull/1161#discussion_r85997021) [](ancestors = 85997021)